### PR TITLE
feat: add argocd app catalog api

### PR DESCRIPTION
**Pull request by balajisa09**
### **User description**
Adding app-catalog api server

___

### **PR Type**
Enhancement, Tests

___

### **Description**
- Implemented a FastAPI-based service to fetch and display ArgoCD application data using Kubernetes API and ArgoCD's CRDs.
- Updated the README to include project description, installation, setup instructions, and environment variable details.
- Added a new Python script `app/main.py` to handle the API logic, including configuration loading, data fetching, and parsing.
- Introduced unittests for the `parse_app_data` function to ensure correct parsing of application data.
- Added necessary dependencies in `requirements.txt` for FastAPI, Kubernetes, and testing.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Update README with project details and setup instructions</code></dd></summary>
<hr>

README.md

<li>Updated project description to reflect new functionality.<br> <li> Added installation and setup instructions.<br> <li> Included environment variable details.<br>

</details>

  </td>
  <td><a href="https://github.com/GlueOps/app-catalog/pull/1/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5">+21/-1</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>main.py</strong><dd><code>Add FastAPI service for ArgoCD application data retrieval</code></dd></summary>
<hr>

app/main.py

<li>Implemented FastAPI service to fetch ArgoCD application data.<br> <li> Added Kubernetes configuration loading with error handling.<br> <li> Created endpoint to return ArgoCD apps with pagination.<br> <li> Implemented logging and exception handling.<br>

</details>

  </td>
  <td><a href="https://github.com/GlueOps/app-catalog/pull/1/files#diff-990f7e4140fab1ad82afc787505090b64d4024e63a891405cd9085e7e6b249dd">+118/-0</a>&nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Dependencies</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>requirements.txt</strong><dd><code>Add project dependencies for FastAPI and Kubernetes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

requirements.txt

- Added dependencies for FastAPI, Kubernetes, and testing.

</details>

  </td>
  <td><a href="https://github.com/GlueOps/app-catalog/pull/1/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>test_parse_app_data.py</strong><dd><code>Add unittests for ArgoCD app data parsing</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests/test_parse_app_data.py

<li>Added unittests for <code>parse_app_data</code> function.<br> <li> Included test cases for various scenarios.<br>

</details>

  </td>
  <td><a href="https://github.com/GlueOps/app-catalog/pull/1/files#diff-2ecd330d16d409df054d55a2ef684327f5a1e4dff828c0ca01a28875978d29d5">+109/-0</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: .gitignore
Comment (venkatamutyala) [inline .gitignore:1]: Could you use an official .gitignore?

https://github.com/github/gitignore/blob/main/Python.gitignore

**Review by venkatamutyala — DISMISSED** [review]

File: app/main.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    
+    try:
+        app_name = app["metadata"]["name"]
+        namespace = app["metadata"]["namespace"]
+        captain_domain = os.getenv("CAPTAIN_DOMAIN")
+        res_app = {
+            "app_name": app_name,
+            "argocd_status": app["status"]["health"]["status"],
+            "last_updated_at": app["status"]["operationState"]["finishedAt"],
+            "app_link": "argocd.{captain_domain}/applications/{namespace}/{app_name}".format(captain_domain=captain_domain, namespace=namespace, app_name=app_name),
```
[the remark this answers is not in the document]
Comment (venkatamutyala) [reply app/main.py:91]: @balajisa09 this looks like it would be a small change to keep things consistent with f-strings that you have used elsewhere.

**Review by venkatamutyala — DISMISSED** [review]

**Review by venkatamutyala — CHANGES_REQUESTED** [review]
I think the load_kube_config is something that i added to the helpers library a while back. Can you take a quick look and see if it makes sense to use mine over yours?

File: app/main.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+# Initialize FastAPI app and logger
+app = FastAPI()
+logger = glueops.setup_logging.configure(level=os.environ.get('LOG_LEVEL', 'WARNING'))
+
+@app.exception_handler(Exception)
+async def generic_exception_handler(request: Request, exc: Exception):
+    logger.error(f"Unhandled exception: {exc}")
+
+@contextmanager
+def load_kube_config():
```
[the remark this answers is not in the document]
Comment (balajisa09) [reply app/main.py]: yeah we can reuse it here

**Review by venkatamutyala — APPROVED** [review]

**Pull request merged**